apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/BFD-2572-Cleanup-Medicare-Beneficairyid-History/invoke-psql.sh

```diff
@@ -0,0 +1,346 @@
+
+if [ -z "${PGHOST}" ] || [ -z "${PGUSER}" ]; then
+    echo "*****  E r r o r - Missing required variables  *****"
+    echo "$0 requires ENV variable(s) for: DB_HOST, DB_USER";
+    echo "or";
+    echo "$0 requires cmd-line args for: <db script> <db host> <db username>";
+    exit 1;
+fi
+
+# set to true if the SQL being executed should be displayed
+SHOW_SQL=false
+
+# STEP_TWO_SETUP_TEMP_SQL
+# ==============================================================
+# sql to create a temp medicare_beneficiaryid_history table that
+# we'll use for the work of analyzing and merging data not found.
+# It doesn't matter since we'll be dropping both the real
+# medicare_beneficiaryid_history_temp table as well as the
+# medicare_beneficiaryid_history table.
+# ==============================================================
+read -r -d '' STEP_TWO_SETUP_TEMP_SQL << EOM
+DROP TABLE IF EXISTS PUBLIC.MEDICARE_BENEFICIARYID_HISTORY_TEMP CASCADE;
+
+CREATE TABLE PUBLIC.MEDICARE_BENEFICIARYID_HISTORY_TEMP AS
+SELECT BENE_MBI_ID,
+	BENE_ID,
+	MBI_NUM ,
+	MBI_EFCTV_BGN_DT,
+	MBI_EFCTV_END_DT,
+	LAST_UPDATED
+FROM MEDICARE_BENEFICIARYID_HISTORY;
+
+ALTER TABLE IF EXISTS public.medicare_beneficiaryid_history_temp
+    ADD CONSTRAINT medicare_beneficiaryid_history_temp_pkey PRIMARY KEY (bene_mbi_id);
+	
+CREATE INDEX IF NOT EXISTS medicare_beneficiaryid_history_temp_bene_id_idx
+    ON public.medicare_beneficiaryid_history_temp USING btree
+    (bene_id ASC NULLS LAST);
+	
+CREATE INDEX IF NOT EXISTS medicare_beneficiaryid_history_temp_mbi_num_idx
+    ON public.medicare_beneficiaryid_history_temp USING btree
+    (mbi_num ASC NULLS LAST);
+EOM
+
+# STEP_THREE_CLEANUP_SQL
+# ==============================================================
+# SQL to delete data from the medicare_beneficiaryid_history_temp
+# table that is already stored in either the beneficiaries table or
+# the beneficiaries_history table.
+# ==============================================================
+read -r -d '' STEP_THREE_CLEANUP_SQL << EOM
+DELETE
+FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP F
+WHERE F.BENE_MBI_ID in (
+-- matches between beneficiaries and medicare_beneficiaryid_history_temp will be deleted
+--
+SELECT A.BENE_MBI_ID
+      FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP A,
+        BENEFICIARIES B
+      WHERE B.BENE_ID = A.BENE_ID
+        AND B.MBI_NUM = A.MBI_NUM
+        AND B.EFCTV_BGN_DT = A.MBI_EFCTV_BGN_DT
+        AND (B.EFCTV_END_DT IS NULL
+                  OR B.EFCTV_END_DT = A.MBI_EFCTV_END_DT)
+UNION
+-- matches between beneficiaries_history and medicare_beneficiaryid_history_temp will be deleted
+--
+SELECT A.BENE_MBI_ID
+      FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP A,
+        BENEFICIARIES_HISTORY B
+      WHERE B.BENE_ID = A.BENE_ID
+        AND B.MBI_NUM = A.MBI_NUM
+        AND B.EFCTV_BGN_DT = A.MBI_EFCTV_BGN_DT
+        AND (B.EFCTV_END_DT IS NULL
+                  OR B.EFCTV_END_DT = A.MBI_EFCTV_END_DT) );
+EOM
+
+# STEP_FOUR_CLEANUP_SQL
```

Comment:
Step 3 and Step 4 have the same summary, could we add some more details to one to describe why it's different from the other?

Reply:
I don't see that; comments are slightly different. In addition, I find putting the real comment in the SQL more valuable, especially for folks that can look at the SQL scripts in the db/scripts repo directory and see what the SQL is actually doing. Putting all comments in the shell script doesn't really help folks with SQL.

Reply:
I agree with @billie-rose. Lines 77-79 and 110-112 look identical to me. Maybe line 75 ought to read something like:

```
# STEP_THREE_CLEANUP_SQL and STEP_FOUR_CLEANUP_SQL
```

And we could, in-turn remove lines 109-113, and move on.

Reply:
done.